store/poi_test.go

```diff
@@ -819,6 +852,58 @@ func (s *POITestSuite) TestGetResourceListImportant() {
 	s.Len(list, len(importantResourceID))
 }
 
+func (s *POITestSuite) TestListPOIByResourceNotAdded() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+	location := schema.Location{
+		Latitude:  25.12345,
+		Longitude: 119.12345,
+	}
+
+	pois, err := store.ListPOIByResource("resource_not_added", location)
+	s.NoError(err)
+	s.Len(pois, 0)
+}
+
+func (s *POITestSuite) TestListPOIByResourceAddedButNoRating() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+	location := schema.Location{
+		Latitude:  25.12345,
+		Longitude: 119.12345,
+	}
+
+	pois, err := store.ListPOIByResource("resource_no_rating", location)
+	s.NoError(err)
+	s.Len(pois, 0)
+}
+
+func (s *POITestSuite) TestListPOIByResourceNormal() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+	location := schema.Location{
+		Latitude:  25.12345,
+		Longitude: 119.12345,
+	}
+
+	pois, err := store.ListPOIByResource("resource_11", location)
+	s.NoError(err)
+	s.Len(pois, 2)
+
+	pois, err = store.ListPOIByResource("resource_12", location)
+	s.NoError(err)
+	s.Len(pois, 1)
+}
+
+func (s *POITestSuite) TestListPOIByResourceOutOfRange() {
+	store := NewMongoStore(s.mongoClient, s.testDBName)
+	location := schema.Location{
+		Latitude:  20,
+		Longitude: 120,
+	}
+
+	pois, err := store.ListPOIByResource("resource_1", location)
```

Comment:
Only `officialResourcesPOIID` provides `resource_1` and its location is at (0, 0) right?